src/app/static/src/app/utils.ts

```diff
@@ -162,6 +162,26 @@ const flattenOption = (filterOption: NestedFilterOption): NestedFilterOption =>
     return result;
 };
 
+export const flattenOptionsIdsByHierarchy = (filterOptions: NestedFilterOption[]): string[] => {
+    const result: string[] = [];
+    const recursive = (filterOptions: NestedFilterOption[]) => {
+        // 1. Push ids for the top level of options
+        filterOptions.forEach((option: NestedFilterOption) => result.push(option.id));
+        let nextLayer: NestedFilterOption[] = [];
+        // 2. Get all options at next layer and recurse
+        filterOptions.forEach((option: NestedFilterOption) => {
```

Comment:
Why are you looping twice ? you could perform the action in a loop, which is more efficient.